realm/realm-library/src/objectServer/java/io/realm/internal/objectstore/OsAppCredentials.java

```diff
@@ -24,54 +30,70 @@
 
     private static final int TYPE_ANONYMOUS = 1;
     private static final int TYPE_API_KEY = 2;
-    private static final int TYPE_APPLE = 3;
-    private static final int TYPE_CUSTOM_FUNCTION = 4;
-    private static final int TYPE_EMAIL_PASSWORD = 5;
-    private static final int TYPE_FACEBOOK = 6;
-    private static final int TYPE_GOOGLE = 7;
-    private static final int TYPE_JWT = 8;
+    private static final int TYPE_SERVER_API_KEY = 3;
+    private static final int TYPE_APPLE = 4;
+    private static final int TYPE_CUSTOM_FUNCTION = 5;
+    private static final int TYPE_EMAIL_PASSWORD = 6;
+    private static final int TYPE_FACEBOOK = 7;
+    private static final int TYPE_GOOGLE = 8;
+    private static final int TYPE_JWT = 9;
     private static final long finalizerPtr = nativeGetFinalizerMethodPtr();
 
     public static OsAppCredentials anonymous() {
-        return new OsAppCredentials(nativeCreate(TYPE_ANONYMOUS));
+        return new OsAppCredentials(nativeCreate(TYPE_ANONYMOUS), Credentials.IdentityProvider.ANONYMOUS);
```

Comment:
I think it would be cleaner if `Credentials.IdentityProvider.ANONYMOUS` was defined in the factory constructor of `AppCredentials`. `OsAppCredentials` is there to map to the underlying type, but us making a shortcut is not the responsibility of the Os class but our own `AppCredentials`. The same everywhere else.